src/devices/Mcp3xxx/Mcp3xxx.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.Spi;
+
+namespace Iot.Device.Adc
+{
+    /// <summary>
+    /// Mcp3xxx Abstract class representing the MCP ADC devices.
+    /// </summary>
+    public abstract class Mcp3xxx : IDisposable
+    {
+        /// <summary>
+        /// InputType: the type of pin connection
+        /// </summary>
+        protected enum InputType
+        {
+            ///<summary>The value is measured as the voltage on a single pin</summary>
+            SingleEnded = 0,
+            ///<summary>The value is the difference in voltage between two pins with the first pin being the positive one</summary>
+            Differential = 1,
+            ///<summary>The value is the difference in voltage between two pins with the second pin being the positive one</summary>
+            InvertedDifferential = 2
+        }
+
+        private SpiDevice _spiDevice;
+
+        /// <summary>
+        /// Constructs Mcp3Adc instance
+        /// </summary>
+        /// <param name="spiDevice">Device used for SPI communication</param>
+        public Mcp3xxx(SpiDevice spiDevice)
+        {
+            _spiDevice = spiDevice;
+        }
+
+        /// <summary>
+        /// Disposes Mcp3Adc instances
+        /// </summary>
+        public void Dispose()
+        {
+            _spiDevice?.Dispose();
+            _spiDevice = null;
+        }
+
+        /// <summary>
+        /// Reads a value from the device
+        /// </summary>
+        /// <param name="adcRequest">A bit pattern to be sent to the ADC.</param>
+        /// <param name="adcRequestLengthBytes">The number of bytes to be sent to the ADC containing the request and returning the response.</param>
+        /// <param name="adcResolutionBits">The number of bits in the returned value</param>
+        /// <param name="delayBits">The number of bits to be delayed between the request and the response being read.</param>
+        /// <returns>A value corresponding to a voltage level on the input pin described by the request.</returns>
+        protected int Read(int adcRequest, int adcRequestLengthBytes, int adcResolutionBits, int delayBits)
```

Comment:
optional: I think it might make sense to rename this method to something else (i.e. ReadInternal) or something along these lines to make it clear it's only meant to be used by implementation